vmware-tanzu/tanzu-framework
# Implement Azure credential update

### What this PR does / why we need it

### Which issue(s) this PR fixes


Fixes #TKG-13946

### Describe testing done for PR


1. Create azure workload cluster with the same AzureSecretIdentity as Management cluster
- Run "tanzu cluster credentials update" to update the workload cluster azure secret: it should return warning for its crendential couldn't be updated seperately
- Run "tanzu management-cluster credentials update" to update the management cluster only: the workload cluster credential is updated correspondingly with the management cluster

2. Create azure workload cluster with different AzureSecretIdentity
- Run "tanzu cluster credentials update" to update the workload cluster azure secret: the crendential could be updated as expected
- Run "tanzu management-cluster credentials update" to update the management cluster only: the workload cluster credential isn't updated while the management cluster credential is updated.

3. Run "tanzu management-cluster credentials update --cascading": all workload clusters crendentials are updated correspondingly with the management cluster


### Release note

```release-note

```



### Additional information

#### Special notes for your reviewer
